integration-tests/src/test/java/tachyon/client/keyvalue/KeyValuePartitionIntegrationTest.java

```diff
@@ -76,4 +78,62 @@ public void readerWriterTest() throws IOException, TachyonException {
     Assert.assertArrayEquals(VALUE2, mKeyValuePartitionReader.get(KEY2));
     Assert.assertNull(mKeyValuePartitionReader.get("NoSuchKey".getBytes()));
   }
+
+  /**
+   * Tests that {@link KeyValuePartitionReader#size()} is correct when a new reader is created.
+   *
+   * @throws Exception when any exception happens
+   */
+  @Test
+  public void sizeTest() throws Exception {
+    byte[][] keys = new byte[][]{KEY1, KEY2};
+    byte[][] values = new byte[][]{VALUE1, VALUE2};
+    for (int size = 0; size <= 2; size ++) {
+      String uniqPath = PathUtils.uniqPath();
+      TachyonURI uri = new TachyonURI(uniqPath);
+      mKeyValuePartitionWriter = KeyValuePartitionWriter.Factory.create(uri);
+      for (int i = 0; i < size; i ++) {
+        mKeyValuePartitionWriter.put(keys[i], values[i]);
+      }
+      mKeyValuePartitionWriter.close();
+
+      mKeyValuePartitionReader = KeyValuePartitionReader.Factory.create(uri);
+      Assert.assertEquals(size, mKeyValuePartitionReader.size());
+      mKeyValuePartitionReader.close();
+    }
+  }
+
+  /**
+   * Tests that {@link KeyValuePartitionReader#iterator()} can iterate over a partition correctly.
+   *
+   * @throws Exception when any exception happens
+   */
+  @Test
+  public void iteratorTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    TachyonURI uri = new TachyonURI(uniqPath);
+    mKeyValuePartitionWriter = KeyValuePartitionWriter.Factory.create(uri);
+    mKeyValuePartitionWriter.put(KEY1, VALUE1);
```

Comment:
I feel testing this on only two keyvalue pairs is probably not idea. can we do it with more items? (in that case, you probably want to automatically generate input keyvalue pairs)